### PR TITLE
Finetuning batrachian mutation (with intended sideeffects)

### PR DESCRIPTION
**Pull request by Stadler76**
#### Summary
Content "Finetuning batrachian mutation (with intended sideeffects)"

#### Purpose of change
The description of the Webbed Hands aka `WEBBED`-mutation read "Your hands and feet are heavily webbed, ..." but the name of the mutation is 'Webbed Hands'. Now they are split into Webbed Hands and Webbed Feet, especially since it makes sense that wearing footwear should reduce the swimming bonus at least slightly since you now can only use your hands to support swimming unless you take off your shoes. Additionaly batrachian now has the `FROG_EYES`-mutation which is a combination of `MEMBRANE`, `SEESLEEP` and `URSINE_EYE`. In Extra Mutated Scenarios you can now choose post-thresh Batrachian.

#### Describe the solution
I first proposed to rename 'Webbed Hands' to 'Webbed Hands and Feet', but after a brief discussion it was suggested to split them up into two mutations.
Note, that I've omitted `"types": [ "FEET" ],` since they should be an addition to the feet, not a replacement. Hence the `"cancels":  [ "WEBBED_FEET" ],` on the `DRAGON_FEET`-mutation. Black Dragons aren't designed to be good swimmers. At least not yet.
And note, that I couldn't add `"cancels":  [ "WEBBED_FEET" ],` to `TOE_CLAWS` or that would break lizard mutation with Magiclysm enabled.
And most importantly: Changes to Magiclysm are side effects. This PR focuses on vanilla.

To fix batrachian causing to fail the breach power test again it was suggested by @Maleclypse to remove batrachian from `URSINE_EYE` and add `MYOPIC_IN_LIGHT` the `EYEBULGE` and in the end I've decided to 'borrow' parts of the work in progress mutatant limbs pt 1 PR from @Venera3 to add `FROG_EYES` instead.

#### Describe alternatives you've considered
- Just rename the mutation and don't split it (See above, why I didn't do that).
- ~~Swap the numbers for feet and hands, so that the feet add more weight to the swimming bonus.~~ [Edit:] BTDT: See b6989056d63400fedf3b13fcb4d6942de844538a.

#### Testing
Did some brief tests with lizards and dragons with and lizards without Magiclysm. All is working as intended and dragons are not affected. Meaning: They don't obtain Webbed Feet and they lose them if they've obtained them from elsewhere.
After adding `FROG_EYES` to the mix I've tested batrachian with and without Magiclysm and it works as intended (I think?).

### CODE REVIEW

**Comment by Maleclypse** [comment]
This makes sense.  Would it make more sense to jsonize swim speed modifier and remove both of them from character?   If you don't know how to jsonize things no problem, I don't either :)

**Comment by Stadler76** [comment]
> This makes sense. Would it make more sense to jsonize swim speed modifier and remove both of them from character? If you don't know how to jsonize things no problem, I don't either :)

Yes and yes: Yes, it would make sense to jsonize it, especially since I don't have to rerun the compiler every time I apply a change the the mechanics and yes, I have zero experience on jsonizing things, too 😄

PS: I intend to change the number from 70:30 to 50:50 once the required checks have finished. Feet should have at least the same weight on swimming bonus as the hands. I first only considered human feet aka short toes without thinking of frog aka batrachian mutation. But even, if you're a human with only webbed hands and feet it impllies, that that player/NPC has noticeably longer toes (and shorter feet) compared to a normal human.

File: src/character.cpp
Comment (Stadler76) [inline src/character.cpp:1290]:
```suggestion
        ret -= hand_bonus_mult * webbing_factor * 0.5f;
    }
    if( has_trait( trait_WEBBED_FEET ) ) {
        ret -= ( 1 - footwear_factor() ) * webbing_factor * 0.5f;
```
Should be 50:50 aka minimum weight for the feet should be 50%

**Comment by Maleclypse** [comment]
The root problem is that Batrachian has too many things leading to it's core mutations.

**Comment by Stadler76** [comment]
> The root problem is that Batrachian has too many things leading to it's core mutations.
> 
> So what is really needed is stripping down Batrachian in some ways. Make more of it's post thresh mutations freestanding?

Or how about adding `FROG_EYES` as post-thresh and remove `BATRACHIAN` from `MEMBRANE` and `URSINE_VISION`? And mix in `"flags": [ "MYOPIC_IN_LIGHT" ],`? Similar to what you've [suggested on the Discord](https://discord.com/channels/598523535169945603/598535827169083403/999434902648914050)?

**Comment by Stadler76** [comment]
Ok. This is the first (and final for this PR?) version of the `FROG_EYES`-mutation. I hope you don't mind, that I borrowed parts of it from your mutant limbs pt1 PR, @Venera3. Well, it's just those parts that are easy to backport without breaking more stuff than it fixes. e. g. I didn't clone the NV from Ursine Vision although that one is not a big loss. First off, batrachian gets Night Vision anyway and second: What's the point of a toggleable NV trait anyway? There's not cost for leaving it on.

Then I've touched another mod: I've added `WEBBED_FEET` to `extra mutation scenarios` as this was pretty straightforward.

**Comment by Stadler76** [comment]
After looking a bit more at `FROG_EYES` from #53004 I've decided to include `SEESLEEP` aka `Lidless Eyes` in them and remove Batrachian from it. Since I've used most, but not all parts from FROG_EYES of that WIP PR you should mostly credit Venera3 for the Frog Eyes. This saves you 100 more mutagen from going toward batrachian threshold.

They still don't contain bulging eyes since I don't want to touch `mutation_limbs.json` aka the body-parts system. Mainly because I think that backporting it to the current version would break more, than it fixes. And it would be kinda half-assed to do cherry-picking for a single addition.

Any feedback on that is welcome and if undesired it can easily be reverts. But I'd leave the changes to `src/character_armor.cpp` as a byproduct since that seems like an oversight there when implementing the `SEESLEEP`-flag.

**Comment by Maleclypse** [comment]
#51630  Is where I added the Frog mutation tree and has a ton of comments and discussion of what I was trying to do.

**Comment by Maleclypse** [comment]
Anyway I've had a chance to look and we really only need to add a Batrachian category in between Alpha and Chimera.  Batrachian has less total breach power than Chimera and yet Chimera still works.  And when you look at the way breach is determined, Alpha doesn't have plusses to breach chance and chimera doesn't have negatives to it's chance so the only thing different about them is that they are tested differently than the other categories.   So anyway that's my suggestion, let me know if you need me to make the Batrachian test category and I'll get to it this weekend.

**Comment by Maleclypse** [comment]
```GIVEN( "The player has all pre-threshold mutations for " + cat_id.str() ) {
            npc dummy;
            dummy.set_body();
            dummy.give_all_mutations( cur_cat, false );

            const int breach_chance = dummy.mutation_category_level[cat_id];
            if( cat_id == mutation_category_ALPHA ) {
                THEN( "Alpha Threshold breach power is between 35 and 60" ) {
                    INFO( "MUTATIONS: " << get_mutations_as_string( dummy ) );
                    CHECK( breach_chance >= 35 );
                    CHECK( breach_chance <= 60 );
                }
                continue;
            } else if( cat_id == mutation_category_CHIMERA ) {
                THEN( "Chimera Threshold breach power is between 100 and 160" ) {
                    INFO( "MUTATIONS: " << get_mutations_as_string( dummy ) );
                    CHECK( breach_chance >= 100 );
                    CHECK( breach_chance <= 160 );
                }
                continue;
            }
            THEN( "Threshold breach power is between 55 and 90" ) {
                INFO( "MUTATIONS: " << get_mutations_as_string( dummy ) );
                CHECK( breach_chance >= BREACH_POWER_MIN );
                CHECK( breach_chance <= BREACH_POWER_MAX );
            }
            ```

**Comment by Stadler76** [comment]
> I added see-sleep as a post thresh ability because "The brains of the frogs when they sleep never really switch off. They remain highly alert to their surroundings and will react to changes in the environment as quickly as they would if they were not asleep. And this is the biggest difference between how frogs sleep and how we sleep." Frogs can basically move from a rest state straight into ambush predator mode instantaneously.

I just wanted to include them into frog eyes since I saw, that Venera3 had the same intentions. 'Fixing' `src/character_armor.cpp` is just a byproduct of that. Turning `has_trait`-checks into `has_flag`-checks was my first PR here. See #48850. And I'm not opposed to do more of that, e. g. `has_trait( trait_COLDBLOOD4 )` → `has_flag( json_flag_ECTOTHERMIC )`. But that would be in a separate PR I'd say.

> #51630 Is where I added the Frog mutation tree and has a ton of comments and discussion of what I was trying to do.
> 
> Not all of it worked at the time.

Yeah, I know. Twas like right after you've spotted that suggestion. And is a nice alternative to fish mutation btw.

> Anyway I've had a chance to look and we really only need to add a Batrachian category in between Alpha and Chimera. Batrachian has less total breach power than Chimera and yet Chimera still works. And when you look at the way breach is determined, Alpha doesn't have plusses to breach chance and chimera doesn't have negatives to it's chance so the only thing different about them is that they are tested differently than the other categories. So anyway that's my suggestion, let me know if you need me to make the Batrachian test category and I'll get to it this weekend.

Not needed for this PR now and I'll go check, if my other PR still breaks after this one is merged, but I'd bet there will be more PRs like this one that will cause it to break again.
btw: Does the breach power test include mods? Because if so I wonder if some undesired decisions have been made for the black dragon mutation. If not, then I haven't said anything.

PS: Next one I'm planning to do is #59362 before I try to get the eyebulge/membrane-split fixed for batrachian.

**Comment by Stadler76** [comment]
how about this:
```cpp
            } else if( cat_id == mutation_category_BATRACHIAN ) {
                THEN( "Batrachian Threshold breach power is between 65 and 110" ) {
                    INFO( "MUTATIONS: " << get_mutations_as_string( dummy ) );
                    CHECK( breach_chance >= 65 );
                    CHECK( breach_chance <= 110 );
                }
                continue;
            }
```
Numbers are the median values between alpha and chimera aka quick and dirty doing the math.
/me hides his calculator between his back ...

**Comment by Stadler76** [comment]
About the changes to Aftershock: I'm about 99% sure, that I did it correctly. Thankfully the maintainers of that mod got pinged to help me with the remaining ~1%.

Note, that there are other mods affected, that I don't wanna touch yet. Mostly, because I'm unsure, what to do there. So feel free to tell me, what to do with CRT_EXPANSION and No Hope.

**Comment by anothersimulacrum** [comment]
CRT_EXPANSION is obsolete, so make no changes to that. @Night-Pryanik maintains No Hope.

**Comment by Stadler76** [comment]
> CRT_EXPANSION is obsolete, so make no changes to that. @Night-Pryanik maintains No Hope.

Thanks for the pointer. And now I'm 99% about No Hope, too.

**Comment by Night-Pryanik** [comment]
Lgtm.

**Comment by Stadler76** [comment]
Some notes about 0d4aec2:
- If I missed any post-thresh mutation for Vatrachian, then tell me.
- I've removed `THRESH_BATRACHIAN` from `MUT_TOUGH` aka Resilient since they can't obtain Resilient currently.
  If you think, that this is a bug, open an issue about that.
- Currently you can't obtain all starting traits for the batrachian mutation.
  This seems to be rather an issue with character creation, than with the mod as it claims that you have conflicting traits which doesn't look, like it's an easy fix.
  Another thing, where I would be happy, if someone opens an issue or PR to fix that.

**Comment by Stadler76** [comment]
@Maleclypse Can you look into the recent changes in commit 0688b58? I'm not in a hurry to get this one merged. On the other hand, if it gets merged there can always be a followup PR. I mainly want to know, if this is in a merge-worthy state.

**Review by Maleclypse — APPROVED** [review]
Thank you for doing this!

**Pull request merged**
